core/src/main/java/com/linecorp/armeria/server/HttpHeaderUtil.java

```diff
@@ -141,32 +147,54 @@ private static InetAddress createInetAddress(String address) throws UnknownHostE
         // Remote quotes. e.g. "[2001:db8:cafe::17]:4711" => [2001:db8:cafe::17]:4711
         final String addr = firstChar == '"' ? QUOTED_STRING_TRIMMER.trimFrom(address) : address;
 
-        if (NetUtil.isValidIpV4Address(addr) || NetUtil.isValidIpV6Address(addr)) {
-            return InetAddress.getByAddress(NetUtil.createByteArrayFromIpAddressString(addr));
+        final InetSocketAddress inetSocketAddress;
+        try {
+            if (addr.charAt(0) == '[') {
+                inetSocketAddress = createInetSocketAddress(addr, ']');
+            } else {
+                inetSocketAddress = createInetSocketAddress(addr, ':');
+            }
+            // Log debug message if it is an invalid address.
+            if (inetSocketAddress == null) {
+                logger.debug("Failed to parse an address: {}", address);
+            }
+            return inetSocketAddress;
+        } catch (NumberFormatException e) {
+            logger.debug("Fail to parse a port from an address: {}", address, e);
+            return null;
         }
+    }
 
-        if (addr.charAt(0) == '[') {
-            final int delim = addr.indexOf(']');
-            if (delim > 0) {
-                // Remove the port part. e.g. [2001:db8:cafe::17]:4711 => [2001:db8:cafe::17]
-                final String withoutPort = addr.substring(0, delim + 1);
-                if (NetUtil.isValidIpV6Address(withoutPort)) {
-                    return InetAddress.getByAddress(NetUtil.createByteArrayFromIpAddressString(withoutPort));
-                }
-            }
+    @Nullable
+    private static InetSocketAddress createInetSocketAddress(String address, char delimiter)
+            throws UnknownHostException {
+        final int delim = address.indexOf(delimiter);
```

Comment:
I think it is not prohibited to use an IPv4 address with brackets, or there might be some weird implementation that does so. What do you think about using `HostAndPort.fromString()` to extract host and port?

Reply:
That is good advice. Let me apply it. 👍

Reply:
> I think it is not prohibited to use an IPv4 address with brackets

I have removed `NetUtil.isValidIpV{4,6}Address()` because I thought the `NetUtil.createByteArrayFromIpAddressString()` validate address string already. 

https://github.com/netty/netty/blob/dcbfe17eeb588a8ba2acb95800776526d01cc5a2/common/src/main/java/io/netty/util/NetUtil.java#L368

But it was my mistake. After your review, I totally understand why the original code needs to check twice. 😊